olive/systems/utils.py

```diff
@@ -31,3 +37,93 @@ def get_model_config(common_args):
             model_json["config"][key] = value
 
     return model_json
+
+
+def get_package_name(execution_provider):
+    PROVIDER_PACKAGE_MAPPING = {
+        "CPUExecutionProvider": "onnxruntime",
+        "CUDAExecutionProvider": "onnxruntime-gpu",
+        "TensorrtExecutionProvider": "onnxruntime-gpu",
+        "OpenVINOExecutionProvider": "onnxruntime-openvino",
+    }
+    return PROVIDER_PACKAGE_MAPPING.get(execution_provider, "onnxruntime")
+
+
+def create_new_system(origin_system, accelerator):
+    PROVIDER_DOCKERFILE_MAPPING = {
+        "CPUExecutionProvider": "Dockerfile.cpu",
+        "CUDAExecutionProvider": "Dockerfile.gpu",
+        "TensorrtExecutionProvider": "Dockerfile.gpu",
+        "OpenVINOExecutionProvider": "Dockerfile.openvino",
+    }
+
+    # create a new system with the same type as the origin system
+    if origin_system.system_type == "LocalSystem":
+        raise NotImplementedError("olive_managed_env is not supported for LocalSystem")
+
+    elif origin_system.system_type == "PythonEnvironment":
+        import tempfile
+        import venv
+
+        from olive.systems.python_environment import PythonEnvironmentSystem
+
+        # Create the virtual environment
+        venv_path = Path(tempfile.TemporaryDirectory(prefix="olive_python_env_").name)
+        venv.create(venv_path, with_pip=True)
+        new_system = PythonEnvironmentSystem(
+            python_environment_path=f"{venv_path}/bin",
+            accelerators=[accelerator.accelerator_type],
+            environment_variables=origin_system.config.environment_variables,
+            prepend_to_path=origin_system.config.prepend_to_path,
+            olive_managed_env=True,
+            requirements_file=origin_system.config.requirements_file,
+        )
+        new_system.install_requirements(accelerator)
```

Comment:
could we move it into the PythonEnvironmentSystem `__init__`?

Reply:
The install_requirements needs execution provider info, which is not included in System configuration.